cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java

```diff
@@ -587,70 +392,59 @@ public void testReconcileUpdatesKafkaStorageType(Params params, VertxTestContext
+                    context.failNow(new Exception("If storage is not persistent, something has gone wrong"));
                 }
-                kafkaAssembly(NAMESPACE, CLUSTER_NAME).patch(PatchContext.of(PatchType.JSON), updatedStorageKafka);
+                kafkaAssembly(NAMESPACE, CLUSTER_NAME).patch(updatedStorageKafka);
 
                 LOGGER.info("Updating with changed storage type");
             })))
             .compose(v -> operator.reconcile(new Reconciliation("test-trigger", Kafka.RESOURCE_KIND, NAMESPACE, CLUSTER_NAME)))
             .onComplete(context.succeeding(v -> context.verify(() -> {
+                int originalSize = client.persistentVolumeClaims().inNamespace(NAMESPACE).list().getItems().size();
                 // Check the Volumes and PVCs were not changed
-                assertPVCs(context, KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), originalPVCs.get());
-                assertVolumes(context, KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), originalVolumes.get());
-                assertInitContainers(context, KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), originalInitContainers.get());
+                assertPVCs(context, CLUSTER_NAME + "-kafka", originalSize);
+                assertVolumes(context, CLUSTER_NAME + "-kafka");
                 async.flag();
             })));
     }
 
-    private void assertPVCs(VertxTestContext context, String statefulSetName, List<PersistentVolumeClaim> originalPVCs) {
-        context.verify(() -> {
-            StatefulSet statefulSet = client.apps().statefulSets().inNamespace(NAMESPACE).withName(statefulSetName).get();
-            assertThat(statefulSet, is(notNullValue()));
-            List<PersistentVolumeClaim> pvcs = statefulSet.getSpec().getVolumeClaimTemplates();
-            assertThat(originalPVCs.size(), is(pvcs.size()));
-            assertThat(originalPVCs, is(pvcs));
-        });
 
-    }
-
-    private void assertVolumes(VertxTestContext context, String statefulSetName, List<Volume> originalVolumes) {
+    private void assertPVCs(VertxTestContext context, String podSetName, int originalSize) {
         context.verify(() -> {
-            StatefulSet statefulSet = client.apps().statefulSets().inNamespace(NAMESPACE).withName(statefulSetName).get();
-            assertThat(statefulSet, is(notNullValue()));
-            List<Volume> volumes = statefulSet.getSpec().getTemplate().getSpec().getVolumes();
-            assertThat(originalVolumes.size(), is(volumes.size()));
-            assertThat(originalVolumes, is(volumes));
+            StrimziPodSet sps = supplier.strimziPodSetOperator.client().inNamespace(NAMESPACE).withName(podSetName).get();
+            assertThat(sps, is(notNullValue()));
+            List<PersistentVolumeClaim> pvc = client.persistentVolumeClaims().inNamespace(NAMESPACE).list().getItems();
+            assertThat(pvc.size(), is(originalSize));
```

Comment:
This is not exactly nice since you:

Get the PodSet and check its not null ... Why? Does not seem to belong here.
Check all PVCs for single size ... you should check only the PVCs which belong to this PodSet

Reply:
I have pushed the changes for the suggestion but I am little confused on getting PVC corresponding to the PodSet.

Reply:
You can for example filter it by name => they will start with the name of the pod / podset.